receiver/hostmetricsreceiver/internal/metadata/generated_metrics.go

```diff
@@ -23,42 +23,74 @@ import (
 
 // Type is the component type name.
 const Type configmodels.Type = "hostmetricsreceiver"
+const SystemCPUTime = "system.cpu.time"
+const SystemMemoryUsage = "system.memory.usage"
```

Comment:
Would like to get rid of these if we could and have `Metrics.SystemCPUTime.Name()` be the canonical way to get the metric name. Since we have metrics and labels all in one `metadata` package it's not clear what these variables are (metrics or labels).

For below I guess you could do:

```go
var MetricsByName = map[string]metricIntf{
	Metrics.SystemCPUTime.Name():     Metrics.SystemCPUTime,
	Metrics.SystemMemoryUsage.Name(): Metrics.SystemMemoryUsage,
}	
```

(Hopefully Go would inline all those function calls but since this is code-generated could just stick the literal string in there).

or is this too verbose?

Reply:
Why can't the module-level variables just be the interface with `Name`, `New`, `Labels`, etc?  I.e. Do we really need to nest everything in a structure called `Metrics`, or can we just flatten it out to make things less verbose?  I doubt there would ever be naming conflicts with the other variable names and the type system would deal with any misuse.  So it seems like unless there is a reason to pass around the `Metrics` struct as a whole it doesn't serve much purpose other than namespacing the metric fields.

Or alternatively maybe `Metrics` could just be called `M`, similar to how the testing package calls things `T` and `B` as a well understood convention.

I just want to keep the length of things down since in client code the reference would be `metadata.Metrics.SystemMemoryUsage.Name()` which seems a bit long.

Reply:
Would `metadata.Metrics.SystemMemoryUsage.Name()` be used that much or would `metadata.Metrics.SystemMemoryUsage.Create()`? (Our lint rules don't currently allow it but if you were able to import the `metadata` package as `import . "github.com/.../metadata"` then we could do `Metrics.SystemMemoryUsage`.)

I wanted to provide some organization between metrics and labels. For example if we wanted a way to get a list of all metrics I was thinking of having `metadata.Metrics.All() -> []Metric`. The other alternative was to have a package under `metadata` called `metrics` and `labels` (though would have to make sure you don't run into package cycles) so that you would have `metrics.SystemMemoryUsage`. 

Could the builder library take an interface so you could pass in `metadata.Metrics.SystemMemoryUsage` that adheres to an interface so you wouldn't have to call .Name() or .Create()?

Reply:
Ok, I did another round that gets rid of the constant name strings and uses the `Name()` methods, but also adds an alias for `M = Metrics` and `L = Labels`.  That should help some with verbosity.

I would prefer shortening the package name to `meta` or something before using the `import .` thing so that it is more explicit.

Reply:
Looks good. For the metadata thing people can always do 

```go
import m "github.com/.../metadata"
```

If it's too long so that part I'm not too worried about.